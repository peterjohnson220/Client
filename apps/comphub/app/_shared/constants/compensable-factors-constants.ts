export class CompensableFactorsConstants {
  public static SKILLS = 'Skills';
  public static CERTS = 'Certs';
  public static YEARS_EXPERIENCE = 'Years_Experience';
  public static EDUCATION = 'Education';
  public static SUPERVISORY_ROLE = 'Supervisory_Role';
}
