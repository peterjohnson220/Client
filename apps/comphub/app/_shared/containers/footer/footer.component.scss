@import 'comphub-page';
@import 'comphub-styles';
@import 'assets/scss/colors';

button {
  height: 38px;
}

.cancel-btn {
  font-size: 16px;
  box-shadow: none;
}

.actions-section {
  justify-content: flex-start;
  background-color: white;
  padding-bottom: 0;
  z-index: 99999;
  position: relative;
  box-shadow: 0 -1px 3px 0 rgba(0,0,0,0.17);
}

.smb-counter {
  font-weight: 500;

  .count {
    font-size: 1.5rem;
    margin-right: .5rem;
  }
}

.smb-limit-disabling-message {
  position: absolute;
  color: white;
  display: flex;
  bottom: 65px;
  margin: 0 auto;
  width: 500px;
  padding: 1rem;
  border-radius: .25rem;
  background-color: $pf-additional-colors-green;
  left: 0px;
  right: 0px;

  .icon {
    font-size: 2rem;
    display: flex;
    align-items: center;
    justify-content: center;
    margin-right: 1.5rem;
  }

  a {
    color: white;
  }
}
