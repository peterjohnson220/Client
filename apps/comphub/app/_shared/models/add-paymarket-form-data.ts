export interface AddPayMarketFormData {
  Name: string;
  Country: string;
  Currency: string;
  Location: string;
  Industry: string;
  Size: string;
  GeoLabel: string;
  OrganizationTypeId: number;
  IsGovernmentContractor: boolean;
}

export function generateMockAddPayMarketFormData(): AddPayMarketFormData {
  return {
    Name: 'Pay Market Name',
    Country: 'USA',
    Currency: 'USD',
    Location: 'Burlington, MA',
    Industry: 'Software',
    Size: '100 - 500',
    GeoLabel: 'CityState',
    OrganizationTypeId: 1,
    IsGovernmentContractor: false
  };
}
