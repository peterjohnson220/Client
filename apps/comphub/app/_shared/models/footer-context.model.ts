import { ComphubPages } from '../data';

export interface FooterContext {
  HideBackButton: boolean;
  BackButtonEnabled: boolean;
  HideNextButton: boolean;
  NextButtonEnabled: boolean;
  PageTitle: string;
  PreviousPageTitle: string;
  NextPageTitle: string;
  DisplayCancelButton: boolean;
  CancelButtonTitle?: string;
  DisplaySaveButton: boolean;
  SaveButtonEnabled: boolean;
}

export interface FooterContextRequest {
  PageId: ComphubPages;
  JobPricingBlocked: boolean;
  JobSelected: boolean;
  JobDataSelected: boolean;
  IsPeerComphubType: boolean;
  ShowJobPricedHistorySummary: boolean;
  SmbLimitReached: boolean;
  SelectedTrendId: number;
}

export class FooterHelper {
  static jobsFooterContext(nextButtonEnabled: boolean, cancelButtonDisplay: boolean): FooterContext {
    return {
      HideBackButton: true,
      BackButtonEnabled: false,
      HideNextButton: false,
      NextButtonEnabled: nextButtonEnabled,
      PageTitle: 'Jobs',
      PreviousPageTitle: null,
      NextPageTitle: 'Markets',
      DisplayCancelButton: cancelButtonDisplay,
      DisplaySaveButton: false,
      SaveButtonEnabled: false
    };
  }
  static marketsFooterContext(isPeerComphubType: boolean): FooterContext {
    return {
      HideBackButton: false,
      BackButtonEnabled: true,
      HideNextButton: false,
      NextButtonEnabled: true,
      PageTitle: 'Markets',
      PreviousPageTitle: 'Jobs',
      NextPageTitle: isPeerComphubType ? 'Data' : 'Summary',
      DisplayCancelButton: true,
      DisplaySaveButton: false,
      SaveButtonEnabled: false
    };
  }
  static dataFooterContext(jobDataSelected: boolean): FooterContext {
    return {
      HideBackButton: false,
      BackButtonEnabled: true,
      HideNextButton: false,
      NextButtonEnabled: jobDataSelected,
      PageTitle: 'Data',
      PreviousPageTitle: 'Markets',
      NextPageTitle: 'Summary',
      DisplayCancelButton: true,
      DisplaySaveButton: false,
      SaveButtonEnabled: false
    };
  }
  static summaryFooterContext(isPeerComphubType: boolean, showJobPricedHistorySummary: boolean, smbLimitReached: boolean): FooterContext {
    return {
      HideBackButton: isPeerComphubType || showJobPricedHistorySummary,
      BackButtonEnabled: !showJobPricedHistorySummary,
      HideNextButton: true,
      NextButtonEnabled: false,
      PageTitle: 'Summary',
      PreviousPageTitle: isPeerComphubType ? 'Data' : 'Markets',
      NextPageTitle: null,
      DisplayCancelButton: smbLimitReached,
      CancelButtonTitle: smbLimitReached ? 'Close' : 'Cancel',
      DisplaySaveButton: false,
      SaveButtonEnabled: false
    };
  }

  static trendsLandingFooterContext(trendIdIsSelected: boolean) {
    return {
      HideBackButton: true,
      BackButtonEnabled: false,
      HideNextButton: trendIdIsSelected,
      NextButtonEnabled: true,
      PageTitle: 'Landing',
      PreviousPageTitle: null,
      NextPageTitle: 'Jobs',
      DisplayCancelButton: false,
      CancelButtonTitle: null,
      DisplaySaveButton: false,
      SaveButtonEnabled: false
    };
  }

  static trendsJobsFooterContext() {
    return {
      HideBackButton: false,
      BackButtonEnabled: true,
      HideNextButton: false,
      NextButtonEnabled: true,
      PageTitle: 'Jobs',
      PreviousPageTitle: 'Landing',
      NextPageTitle: 'Scopes',
      DisplayCancelButton: false,
      CancelButtonTitle: null,
      DisplaySaveButton: false,
      SaveButtonEnabled: false
    };
  }

  static trendsScopesFooterContext() {
    return {
      HideBackButton: false,
      BackButtonEnabled: true,
      HideNextButton: false,
      NextButtonEnabled: true,
      PageTitle: 'Scopes',
      PreviousPageTitle: 'Jobs',
      NextPageTitle: 'Summary',
      DisplayCancelButton: false,
      CancelButtonTitle: null,
      DisplaySaveButton: false,
      SaveButtonEnabled: false
    };
  }

  static trendsSummaryFooterContext(trendIdIsSelected: boolean) {
    return {
      HideBackButton: trendIdIsSelected,
      BackButtonEnabled: true,
      HideNextButton: true,
      NextButtonEnabled: false,
      PageTitle: 'Summary',
      PreviousPageTitle: 'Scopes',
      NextPageTitle: null,
      DisplayCancelButton: false,
      CancelButtonTitle: null,
      DisplaySaveButton: !trendIdIsSelected,
      SaveButtonEnabled: true
    };
  }

  static getFooterContext(request: FooterContextRequest): FooterContext {
    switch (request.PageId) {
      case ComphubPages.Jobs:
        return this.jobsFooterContext(this.jobsPageNextButtonEnable(request), request.JobSelected);
      case ComphubPages.Markets:
        return this.marketsFooterContext(request.IsPeerComphubType);
      case ComphubPages.Data:
        return this.dataFooterContext(request.JobDataSelected);
      case ComphubPages.Summary:
        return this.summaryFooterContext(request.IsPeerComphubType, request.ShowJobPricedHistorySummary, request.SmbLimitReached);
      case ComphubPages.TrendsLanding:
        return this.trendsLandingFooterContext(!!request.SelectedTrendId);
      case ComphubPages.TrendsJobs:
        return this.trendsJobsFooterContext();
      case ComphubPages.TrendsScopes:
        return this.trendsScopesFooterContext();
      case ComphubPages.TrendsSummary:
        return this.trendsSummaryFooterContext(!!request.SelectedTrendId);
      default:
        return null;
    }
  }

  private static jobsPageNextButtonEnable(request: FooterContextRequest): boolean {
    return !request.JobPricingBlocked &&
           ((request.IsPeerComphubType && !!request.JobSelected) ||
            (!request.IsPeerComphubType && request.JobDataSelected));
  }
}

export function generateMockFooterContext(): FooterContext {
  return {
    HideBackButton: true,
    BackButtonEnabled: false,
    HideNextButton: false,
    NextButtonEnabled: false,
    PageTitle: 'Jobs',
    PreviousPageTitle: '',
    NextPageTitle: 'Markets',
    DisplayCancelButton: true,
    DisplaySaveButton: false,
    SaveButtonEnabled: false
  };
}
