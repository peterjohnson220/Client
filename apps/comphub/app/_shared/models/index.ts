export * from './add-paymarket-form-data';
export * from './country-data.set';
export * from './footer-context.model';
export * from './job-pricing-limit-info.model';
export * from './job-salary-trend.model';
export * from './market-data-location';
export * from './market-data-scope';
export * from './quick-price-history.model';
export * from './quick-price-grid-column.model';
export * from './quick-price-grid.model';
export * from './trending-job-group.model';
export * from './workflow-context';
