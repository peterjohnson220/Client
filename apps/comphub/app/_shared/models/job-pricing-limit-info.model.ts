export interface JobPricingLimitInfo {
  Used: number;
  Available: number;
}
