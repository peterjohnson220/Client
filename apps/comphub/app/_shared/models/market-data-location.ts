export interface MarketDataLocation {
  LocationName: string;
  GeoLabelDisplayName: string;
  GeoLabel: string;
}
