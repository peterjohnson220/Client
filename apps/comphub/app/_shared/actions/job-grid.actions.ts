import { Action } from '@ngrx/store';

import { JobData, JobGridData } from 'libs/models/comphub';
import { GetCrowdSourcedJobPricingRequest } from 'libs/models/payfactors-api';

import { QuickPriceGridContext } from '../models';


export const GET_QUICK_PRICE_MARKET_DATA  = '[Comphub/Jobs Card] Get Quick Price Data';
export const GET_QUICK_PRICE_DATA_SUCCESS  = '[Comphub/Jobs Card] Get Quick Price Data Success';
export const GET_QUICK_PRICE_MARKET_DATA_ERROR  = '[Comphub/Jobs Card] Get Quick Price Data Error';
export const LOAD_MORE_DATA_SUCCESS = '[Comphub/Jobs Card] Load More Data Success';
export const TOGGLE_JOB_DESCRIPTION = '[Comphub/Jobs Card] Toggle Job Description';
export const GET_PEER_JOB_DATA = '[Comphub/Jobs Card] Get Peer Job Data';
export const SEARCH_CROWD_SOURCED_JOBS_BY_TITLE = '[Comphub/Crowd Sourced Page] Search Crowd Sourced Jobs By Title';
export const SEARCH_CROWD_SOURCED_JOBS_BY_TITLE_SUCCESS = '[Comphub/Crowd Sourced Page] Search Crowd Sourced Jobs By Title Success';
export const SEARCH_CROWD_SOURCED_JOBS_BY_TITLE_ERROR = '[Comphub/Crowd Sourced Page] Search Crowd Sourced Jobs By Title Error';
export const GET_CROWD_SOURCED_JOB_PRICING = '[Comphub/Crowd Sourced Page] Get Crowd Sourced Job Pricing';
export const GET_CROWD_SOURCED_JOB_PRICING_SUCCESS = '[Comphub/Crowd Sourced Page] Get Crowd Sourced Job Pricing Success';
export const GET_CROWD_SOURCED_JOB_PRICING_ERROR = '[Comphub/Crowd Sourced Page] Get Crowd Sourced Job Pricing Error';
export const TOGGLE_CROWD_SOURCED_TASKS = '[Comphub/ Crowd Sourced Page] Toggle Crowd Sourced Tasks';

export class GetQuickPriceMarketData implements Action {
  readonly type = GET_QUICK_PRICE_MARKET_DATA;

  constructor(public payload: QuickPriceGridContext) {}
}

export class GetQuickPriceDataSuccess implements Action {
  readonly type = GET_QUICK_PRICE_DATA_SUCCESS;

  constructor(public payload: JobGridData) {}
}

export class GetQuickPriceMarketDataError implements Action {
  readonly type = GET_QUICK_PRICE_MARKET_DATA_ERROR;
}

export class LoadMoreDataSuccess implements Action {
  readonly type = LOAD_MORE_DATA_SUCCESS;

  constructor(public payload: JobData[]) {}
}

export class ToggleJobDescription implements Action {
  readonly type = TOGGLE_JOB_DESCRIPTION;

  constructor(public payload: { jobId: number }) {}
}

export class GetPeerJobData implements Action {
  readonly type = GET_PEER_JOB_DATA;
}

export class SearchCrowdSourcedJobsByTitle implements Action {
  readonly type = SEARCH_CROWD_SOURCED_JOBS_BY_TITLE;

  constructor(public payload: string) {}
}

export class SearchCrowdSourcedJobsByTitleSuccess implements Action {
  readonly type = SEARCH_CROWD_SOURCED_JOBS_BY_TITLE_SUCCESS;

  constructor(public payload: JobGridData) {}
}

export class SearchCrowdSourcedJobsByTitleError implements Action {
  readonly type = SEARCH_CROWD_SOURCED_JOBS_BY_TITLE_ERROR;

  constructor() {}
}
export class GetCrowdSourcedJobPricing implements Action {
  readonly type = GET_CROWD_SOURCED_JOB_PRICING;

  constructor(public payload: GetCrowdSourcedJobPricingRequest) {}
}

export class GetCrowdSourcedJobPricingSuccess implements Action {
  readonly type = GET_CROWD_SOURCED_JOB_PRICING_SUCCESS;

  constructor(public payload: JobData) {}
}

export class GetCrowdSourcedJobPricingError implements Action {
  readonly type = GET_CROWD_SOURCED_JOB_PRICING_ERROR;

  constructor() {}
}

export class ToggleCrowdSourcedTasks implements Action {
  readonly type = TOGGLE_CROWD_SOURCED_TASKS;

  constructor(public payload: {jobTitle: string}) {}
}

export type Actions
  = GetQuickPriceMarketData
  | GetQuickPriceDataSuccess
  | GetQuickPriceMarketDataError
  | LoadMoreDataSuccess
  | ToggleJobDescription
  | GetPeerJobData
  | SearchCrowdSourcedJobsByTitle
  | SearchCrowdSourcedJobsByTitleSuccess
  | SearchCrowdSourcedJobsByTitleError
  | GetCrowdSourcedJobPricing
  | GetCrowdSourcedJobPricingSuccess
  | GetCrowdSourcedJobPricingError
  | ToggleCrowdSourcedTasks;
