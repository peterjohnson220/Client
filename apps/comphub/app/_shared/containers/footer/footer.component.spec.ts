import { NO_ERRORS_SCHEMA } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { combineReducers, Store, StoreModule } from '@ngrx/store';
import { of } from 'rxjs';


import { WindowRef } from 'libs/core/services';
import * as fromRootState from 'libs/state/state';
import { DojGuidelinesService } from 'libs/features/peer/guidelines-badge/services/doj-guidelines.service';
import { AppConstants } from 'libs/constants';

import * as fromComphubPageActions from '../../actions/comphub-page.actions';
import * as fromComphubSharedReducer from '../../reducers';
import { ComphubFooterComponent } from './footer.component';
import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
import { generateMockFooterContext, generateMockPeerWorkflowContext, generateMockWorkflowContext } from '../../models';
import * as fromLibsExchangeExplorerReducers from 'libs/features/peer/exchange-explorer/reducers';
import * as fromLibsFeatureBasicDataGrid from 'libs/features/grids/basic-data-grid/reducers';

class DojGuidelinesStub {
  passing = true;

  get passesGuidelines(): boolean {
    return this.passing;
  }

  validateDataCut(selections: any) {
    return;
  }
}

describe('Comphub - Shared - Footer', () => {
  let instance: ComphubFooterComponent;
  let fixture: ComponentFixture<ComphubFooterComponent>;
  let store: Store<fromComphubSharedReducer.State>;
  let windowRef: WindowRef;
  let modalService: NgbModal;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        StoreModule.forRoot({
          ...fromRootState.reducers,
          comphub_shared: combineReducers(fromComphubSharedReducer.reducers),
          feature_peer_exchangeExplorer: combineReducers(fromLibsExchangeExplorerReducers.reducers),
          feature_basic_data_grid: combineReducers(fromLibsFeatureBasicDataGrid.reducers)
        })
      ],
      providers: [
        {
          provide: WindowRef,
          useValue: { nativeWindow: { location: jest.fn() } }
        },
        { provide: DojGuidelinesService, useClass: DojGuidelinesStub },
        {
          provide: NgbModal,
          useValue: { open: jest.fn() }
        }
      ],
      declarations: [ ComphubFooterComponent ],
      schemas: [ NO_ERRORS_SCHEMA ]
    });

    const footerContext = {...generateMockFooterContext(), NextButtonEnabled: true};

    fixture = TestBed.createComponent(ComphubFooterComponent);
    instance = fixture.componentInstance;
    instance.footerContext$ = of(footerContext);
    instance.ngOnInit();
    store = TestBed.inject(Store);
    windowRef = TestBed.inject(WindowRef);
    modalService = TestBed.inject(NgbModal);
    fixture.detectChanges();
  });

  it('should dispatch a NavigateToNextCard action, when handling the next button click', () => {
    // arrange
    spyOn(store, 'dispatch');
    const expectedAction = new fromComphubPageActions.NavigateToNextCard();

    instance.handleNextButtonClicked();

    expect(store.dispatch).toHaveBeenCalledWith(expectedAction);
  });

  it('should dispatch a NavigateToPreviousCard action, when handling the back button click', () => {
    spyOn(store, 'dispatch');
    const expectedAction = new fromComphubPageActions.NavigateToPreviousCard();

    instance.handleBackButtonClicked();

    expect(store.dispatch).toHaveBeenCalledWith(expectedAction);
  });

  it('should hide the back button, when told to', () => {
    instance.footerContext.HideBackButton = true;
    instance.footerContext.PageTitle = 'Jobs';

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should hide the next button, when told to', () => {
    instance.footerContext.HideNextButton = true;
    instance.footerContext.PageTitle = 'Jobs';

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should disable next button, when told to', () => {
    instance.footerContext.HideNextButton = false;
    instance.footerContext.NextButtonEnabled = false;
    instance.footerContext.PageTitle = 'Jobs';

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should set the window location to the default dashboard, when handling the confirm close app click for a enterprise quick price type', () => {
    instance.workflowContext = generateMockWorkflowContext();
    instance.handleConfirmedCloseApp();

    expect(windowRef.nativeWindow.location).toBe(`/${AppConstants.HostPath}/dashboard`);
  });

  it('should set the window location to the peer dashboard, when handling the confirm close app click for a peer quick price type', () => {
    instance.workflowContext = generateMockPeerWorkflowContext();
    instance.handleConfirmedCloseApp();

    expect(windowRef.nativeWindow.location).toBe(`/${AppConstants.HostPath}/peer/exchanges/redirect`);
  });
});
