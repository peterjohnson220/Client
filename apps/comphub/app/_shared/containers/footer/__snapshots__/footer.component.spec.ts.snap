// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comphub - Shared - Footer should disable next button, when told to 1`] = `
<pf-comphub-footer
  __ngContext__={[Function LRootView]}
  basicGridStore={[Function Store]}
  comphubPages={[Function Object]}
  countryDataSetsLoaded$={[Function Store]}
  exchangeExplorerStore={[Function Store]}
  footerContext={[Function Object]}
  footerContext$={[Function Observable]}
  footerContextSub={[Function Subscriber]}
  guidelinesService={[Function DojGuidelinesStub]}
  isPeerComphubType="false"
  jobPricingBlocked$={[Function Store]}
  jobPricingLimitInfo="null"
  jobPricingLimitInfo$={[Function Store]}
  jobPricingLimitInfoSub={[Function Subscriber]}
  loadingMap$={[Function Store]}
  loadingPeerMap="false"
  loadingPeerMapSub={[Function Subscriber]}
  modalContent={[Function TemplateRef]}
  nextButtonEnabled="false"
  pricedJobsCount$={[Function Store]}
  saveButtonClicked={[Function EventEmitter_]}
  selectedPageIdDelayed$={[Function Store]}
  smbLimitReached$={[Function Store]}
  store={[Function Store]}
  systemUserGroupNames={[Function Object]}
  userContext$={[Function Store]}
  window={[Function Object]}
  workflowContext={[Function Object]}
  workflowContext$={[Function Store]}
  workflowContextSub={[Function Subscriber]}
>
  <div
    class="actions-section pr-3 align-items-center"
  >
    <div
      class="mr-auto pl-4"
    >
      <div
        class="d-flex align-items-center"
      >
      </div>
    </div>
    <button
      class="btn borderless-blue-btn cancel-btn mt-1 mr-3"
      data-qa-id="btn-close-Jobs"
    >
      <span>
        Cancel
      </span>
    </button>
    <button
      class="btn btn-primary"
      data-qa-id="btn-next-Jobs"
      disabled=""
    >
       Next: Markets 
    </button>
    <span
      class="d-inline-block"
    >
    </span>
  </div>
</pf-comphub-footer>
`;

exports[`Comphub - Shared - Footer should hide the back button, when told to 1`] = `
<pf-comphub-footer
  __ngContext__={[Function LRootView]}
  basicGridStore={[Function Store]}
  comphubPages={[Function Object]}
  countryDataSetsLoaded$={[Function Store]}
  exchangeExplorerStore={[Function Store]}
  footerContext={[Function Object]}
  footerContext$={[Function Observable]}
  footerContextSub={[Function Subscriber]}
  guidelinesService={[Function DojGuidelinesStub]}
  isPeerComphubType="false"
  jobPricingBlocked$={[Function Store]}
  jobPricingLimitInfo="null"
  jobPricingLimitInfo$={[Function Store]}
  jobPricingLimitInfoSub={[Function Subscriber]}
  loadingMap$={[Function Store]}
  loadingPeerMap="false"
  loadingPeerMapSub={[Function Subscriber]}
  modalContent={[Function TemplateRef]}
  nextButtonEnabled="false"
  pricedJobsCount$={[Function Store]}
  saveButtonClicked={[Function EventEmitter_]}
  selectedPageIdDelayed$={[Function Store]}
  smbLimitReached$={[Function Store]}
  store={[Function Store]}
  systemUserGroupNames={[Function Object]}
  userContext$={[Function Store]}
  window={[Function Object]}
  workflowContext={[Function Object]}
  workflowContext$={[Function Store]}
  workflowContextSub={[Function Subscriber]}
>
  <div
    class="actions-section pr-3 align-items-center"
  >
    <div
      class="mr-auto pl-4"
    >
      <div
        class="d-flex align-items-center"
      >
      </div>
    </div>
    <button
      class="btn borderless-blue-btn cancel-btn mt-1 mr-3"
      data-qa-id="btn-close-Jobs"
    >
      <span>
        Cancel
      </span>
    </button>
    <button
      class="btn btn-primary"
      data-qa-id="btn-next-Jobs"
    >
       Next: Markets 
    </button>
    <span
      class="d-inline-block"
    >
    </span>
  </div>
</pf-comphub-footer>
`;

exports[`Comphub - Shared - Footer should hide the next button, when told to 1`] = `
<pf-comphub-footer
  __ngContext__={[Function LRootView]}
  basicGridStore={[Function Store]}
  comphubPages={[Function Object]}
  countryDataSetsLoaded$={[Function Store]}
  exchangeExplorerStore={[Function Store]}
  footerContext={[Function Object]}
  footerContext$={[Function Observable]}
  footerContextSub={[Function Subscriber]}
  guidelinesService={[Function DojGuidelinesStub]}
  isPeerComphubType="false"
  jobPricingBlocked$={[Function Store]}
  jobPricingLimitInfo="null"
  jobPricingLimitInfo$={[Function Store]}
  jobPricingLimitInfoSub={[Function Subscriber]}
  loadingMap$={[Function Store]}
  loadingPeerMap="false"
  loadingPeerMapSub={[Function Subscriber]}
  modalContent={[Function TemplateRef]}
  nextButtonEnabled="false"
  pricedJobsCount$={[Function Store]}
  saveButtonClicked={[Function EventEmitter_]}
  selectedPageIdDelayed$={[Function Store]}
  smbLimitReached$={[Function Store]}
  store={[Function Store]}
  systemUserGroupNames={[Function Object]}
  userContext$={[Function Store]}
  window={[Function Object]}
  workflowContext={[Function Object]}
  workflowContext$={[Function Store]}
  workflowContextSub={[Function Subscriber]}
>
  <div
    class="actions-section pr-3 align-items-center"
  >
    <div
      class="mr-auto pl-4"
    >
      <div
        class="d-flex align-items-center"
      >
      </div>
    </div>
    <button
      class="btn borderless-blue-btn cancel-btn mt-1 mr-3"
      data-qa-id="btn-close-Jobs"
    >
      <span>
        Cancel
      </span>
    </button>
    <span
      class="d-inline-block"
    >
    </span>
  </div>
</pf-comphub-footer>
`;
